src/libexpr/primops/fetchTree.cc

```diff
@@ -321,7 +321,8 @@ static RegisterPrimOp primop_fetchTree({
 
         - `shallow` (Bool, optional)
 
-          Make a shallow clone when fetching the Git tree.
+          Allow fetching from a shallow remote repository.
```

Comment:
The remote repository doesn't need to be shallow. It's the local clone that is shallow.